data/json/items/gunmod/bionicmods.json

```diff
@@ -0,0 +1,17 @@
+[
+  {
+    "id": "bio_powershot",
+    "type": "GUNMOD",
+    "name": "Power shot",
+    "description": "This is a pseudo item -- the builtin part of a fusion blaster for the maximum power firing mode.",
+    "weight": 1,
+    "volume": 0,
+    "price": 85000,
+    "material": [ "steel" ],
+    "symbol": ":",
+    "color": "light_red",
+    "location": "underbarrel",
+    "gun_data": { "skill": "rifle", "range": 30, "ammo_effects": [ "PLASMA", "EXPLOSIVE", "FLAME", "NPC_AVOID" ], "durability": 10 },
```

Comment:
I guess the explosion in front of player happens because you're trying to combine `gun_data` and `ammo_effects` in one field. Try the following:
```
"gun_data": { "skill": "rifle", "range": 30, "durability": 10 }, 
"ammo_effects": [ "PLASMA", "EXPLOSIVE", "FLAME", "NPC_AVOID" ],
```

Reply:
I tried that, but then the gunmod overrides the default firing modes behavior. Which I guess would be handy if you wanted a mod that say, filled your arrowheads with an electric charge before firing them or something gonzo like that.
 But here it just adds the explosive and fire flags to the default single and triple shot modes, and we're back to the gun arm being useless when someone gets close.

Reply:
Looks an awful lot like the "not a bug" that I reported ages back. https://github.com/CleverRaven/Cataclysm-DDA/issues/26415 - if you add a point of damage, it might work.

Reply:
Yeah, that seems to fix it!